docs/aws/audit/iamcompliance/rules/root_account_mfa.mdx

```diff
@@ -23,6 +23,247 @@ HIPAA, GDPR, CISAWS, CBP, NIST, SOC2, PCIDSS, HITRUST, AWSWAF, NISTCSF
+<Accordion title='Using Python'>
+To prevent the misconfiguration of not having Multi-Factor Authentication (MFA) enabled for the root account in AWS IAM using Python scripts, you can follow these steps:
+
+1. **Install AWS SDK for Python (Boto3):**
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already.
+
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Python Script to Check MFA Status:**
+   Write a Python script that uses Boto3 to check if MFA is enabled for the root account. This script will help you identify if the root account does not have MFA enabled.
+
+   ```python
+   import boto3
+
+   def check_root_mfa():
+       client = boto3.client('iam')
+       response = client.get_account_summary()
+       mfa_devices = client.list_mfa_devices(UserName='root')
+       
+       if response['SummaryMap']['AccountMFAEnabled'] == 1 and len(mfa_devices['MFADevices']) > 0:
+           print("MFA is enabled for the root account.")
+       else:
+           print("MFA is NOT enabled for the root account. Please enable it.")
+
+   if __name__ == "__main__":
+       check_root_mfa()
+   ```
+
+3. **Automate the Script Execution:**
+   Schedule the script to run at regular intervals using a task scheduler like cron (Linux) or Task Scheduler (Windows) to ensure continuous monitoring.
+
+   For example, to run the script every day at midnight using cron, you can add the following line to your crontab:
+
+   ```bash
+   0 0 * * * /usr/bin/python3 /path/to/your_script.py
+   ```
+
+4. **Notify Administrators:**
+   Enhance the script to send notifications (e.g., via email or Slack) if MFA is not enabled. This ensures that administrators are alerted immediately and can take action.
+
+   ```python
+   import boto3
+   import smtplib
+   from email.mime.text import MIMEText
+
+   def send_notification(message):
+       # Set up the server and login details
+       smtp_server = 'smtp.example.com'
+       smtp_port = 587
+       smtp_user = 'your_email@example.com'
+       smtp_password = 'your_password'
+       
+       # Create the email content
+       msg = MIMEText(message)
+       msg['Subject'] = 'AWS Root Account MFA Status Alert'
+       msg['From'] = smtp_user
+       msg['To'] = 'admin@example.com'
+       
+       # Send the email
+       with smtplib.SMTP(smtp_server, smtp_port) as server:
+           server.starttls()
+           server.login(smtp_user, smtp_password)
+           server.sendmail(smtp_user, 'admin@example.com', msg.as_string())
+
+   def check_root_mfa():
+       client = boto3.client('iam')
+       response = client.get_account_summary()
+       mfa_devices = client.list_mfa_devices(UserName='root')
+       
+       if response['SummaryMap']['AccountMFAEnabled'] == 1 and len(mfa_devices['MFADevices']) > 0:
+           print("MFA is enabled for the root account.")
+       else:
+           message = "MFA is NOT enabled for the root account. Please enable it."
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>